TestLogGrowthWhenRefreshingPendingCommands is flaky

This test failed 11 out of the last 1000 builds 

<img width="867" alt="Screen Shot 2022-10-12 at 11 21 10 AM" src="https://user-images.githubusercontent.com/6658984/195447390-8f442aca-75fb-41a0-a689-a23acedd4a19.png">

------- Stdout: -------
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:161: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogGrowthWhenRefreshingPendingCommands607815779
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestLogGrowthWhenRefreshingPendingCommands
    client_raft_test.go:1692: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogGrowthWhenRefreshingPendingCommands607815779
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (13.89s)

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/6911410?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

Jira issue: CRDB-20470

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -1614,6 +1614,7 @@ func TestLogGrowthWhenRefreshingPendingCommands(t *testing.T) {
 		}
 		propNode := tc.GetFirstStoreFromServer(t, propIdx).TestSender()
 		tc.TransferRangeLeaseOrFatal(t, *leaderRepl.Desc(), tc.Target(propIdx))
+		tc.WaitForLeaseUpgrade(ctx, t, *leaderRepl.Desc())
 		testutils.SucceedsSoon(t, func() error {
 			// Lease transfers may not be immediately observed by the new
 			// leaseholder. Wait until the new leaseholder is aware.

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -1606,10 +1606,8 @@ func TestLogGrowthWhenRefreshingPendingCommands(t *testing.T) {
 			}
 		}
 		// Determine which node to propose on. Transfer lease to that node.
-		var propIdx int
-		if !proposeOnFollower {
-			propIdx = 0
-		} else {
+		propIdx := 0
+		if proposeOnFollower {
 			propIdx = 1
 		}
 		propNode := tc.GetFirstStoreFromServer(t, propIdx).TestSender()
@@ -1619,13 +1617,9 @@ func TestLogGrowthWhenRefreshingPendingCommands(t *testing.T) {
 			// Lease transfers may not be immediately observed by the new
 			// leaseholder. Wait until the new leaseholder is aware.
 			repl, err := tc.GetFirstStoreFromServer(t, propIdx).GetReplica(leaderRepl.RangeID)
-			if err != nil {
-				t.Fatal(err)
-			}
+			require.NoError(t, err)
 			repDesc, err := repl.GetReplicaDescriptor()
-			if err != nil {
-				t.Fatal(err)
-			}
+			require.NoError(t, err)
 			if lease, _ := repl.GetLease(); !lease.Replica.Equal(repDesc) {
 				return errors.Errorf("lease not transferred yet; found %v", lease)
 			}
@@ -1686,9 +1680,8 @@ func TestLogGrowthWhenRefreshingPendingCommands(t *testing.T) {
 		require.NoError(t, tc.RestartServer(2))
 
 		// The write should now succeed.
-		if err := <-putRes; err != nil {
-			t.Fatal(err)
-		}
+		err := <-putRes
+		require.NoError(t, err.GoError())
 	})
 }
 

